src/main/java/io/cryostat/net/web/http/api/v2/TargetSnapshotPostHandler.java

```diff
@@ -118,33 +122,46 @@ public boolean isAsync() {
+                getConnectionDescriptorFromParams(requestParams);
         HyperlinkedSerializableRecordingDescriptor desc =
                 targetConnectionManager.executeConnectedTask(
-                        getConnectionDescriptorFromParams(requestParams),
+                        connectionDescriptor,
                         connection -> {
-                            IRecordingDescriptor descriptor =
-                                    connection.getService().getSnapshotRecording();
-
-                            String rename =
-                                    String.format(
-                                            "%s-%d",
-                                            descriptor.getName().toLowerCase(), descriptor.getId());
-
-                            RecordingOptionsBuilder recordingOptionsBuilder =
-                                    recordingOptionsBuilderFactory.create(connection.getService());
-                            recordingOptionsBuilder.name(rename);
-
-                            connection
-                                    .getService()
-                                    .updateRecordingOptions(
-                                            descriptor, recordingOptionsBuilder.build());
-
+                            SnapshotMinimalDescriptor snapshot =
+                                    recordingTargetHelper.createSnapshot(connection).get();
+                            String rename = snapshot.getName();
                             return new SnapshotDescriptor(
                                     rename,
-                                    descriptor,
+                                    snapshot.getOriginalDescriptor(),
                                     webServer.get().getDownloadURL(connection, rename),
                                     webServer.get().getReportURL(connection, rename));
                         });
+        String snapshotName = desc.getName();
+
+        boolean verificationSuccessful = false;
+        try {
+            verificationSuccessful =
+                    recordingTargetHelper.verifySnapshot(connectionDescriptor, snapshotName).get();
+        } catch (ExecutionException e) {
+            if (ExceptionUtils.getRootCause(e) instanceof SnapshotCreationException) {
+                throw new ApiException(
+                        500,
+                        String.format(
+                                "An error occured during the creation of snapshot %s.",
+                                snapshotName));
+            }
+            throw e;
+        }
+
+        if (!verificationSuccessful) {
+            return new IntermediateResponse<HyperlinkedSerializableRecordingDescriptor>()
+                    .statusCode(202)
+                    .statusMessage(
+                            "Snapshot failed to create: The resultant recording was unreadable for some reason, possibly due to a lack of Active, non-Snapshot source recordings to take event data from.")
```

Comment:
This is a really, really long status message. I'd rather that be part of the body instead.

Reply:
Since the body has to be a `HyperlinkedSerializableRecordingDescriptor`, what's the best way to go about returning that `String`? 

I could add a `String status` field to the object, which can be populated with `Healthy` for successful recordings and a message like the one above for failed recordings, with all other fields `null` in the latter case. Thoughts?

Reply:
Actually, it will already be included as part of the V2 response body in the `meta` field. Now that I'm looking at that again (`AbstractV2RequestHandler#writeResponse`), there is no way to have the `.meta.status` of the body differ from the HTTP status message. So this is fine as-is.